### PR TITLE
[#3] [iOS] Update app to follow iOS template structure

### LINKED ISSUE #3: [iOS] Update app to follow iOS template structure
## Why

We should update the KMM project for iOS following the iOS template.

## Acceptance Criteria

- Update the KMM project for iOS following the iOS template https://github.com/nimblehq/ios-templates.

### PR DESCRIPTION
**Pull request by phongvhd93**
- Close #3 

## What happened 👀

Adding the `iOS-template` as a submodule of the KMM template to maintain the consistent project structure for iOS module.

## Insight 📝

- Add `iOS-template`  as a submodule of `KMM-template`
- Add a new `make.sh` which calls the `make.sh` in `iOS` folder and add the `shared` pod to pod file of iOS project.
- Add build configurations to `build.gradle` of the `shared` module. The build configuration of `shared.podspec` must match with the build configuration of iOS module.

## Proof Of Work 📹

Screenshot 2023-08-07 at 10 37 50

### CODE REVIEW

File: shared/build.gradle.kts [hunk 1]
```diff
@@ -28,8 +30,13 @@ kotlin {
         framework {
```
Comment (markgravity) [inline shared/build.gradle.kts:30]: Do we need to update this `ios.deploymentTarget = "14.1"` to match with our `Podfile`? 🤔

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "ios"]
+	path = ios
+	url = https://github.com/nimblehq/ios-templates.git
```
Comment (markgravity) [inline .gitmodules:3]: I prefer to set up it by version, because a new update from `main` could break this kmm template  🙏

Comment (phongvhd93) [reply .gitmodules:3]: When I initiated the submodule, it was tied to the commit hash  `8336744cc7f66001c9ed24018ff2f545ff9a3c5a`. You can see the commit hash of the `ios` module right below. 

Screenshot 2023-08-08 at 18 12 53

File: shared/build.gradle.kts (hunk 1 above)
Comment (phongvhd93) [reply shared/build.gradle.kts:30]: The podfile of the iOS template has a [bug](https://github.com/nimblehq/ios-templates/issues/509) here. I will create a task to update the minimum iOS version of the development pod with Mustache.

Comment (markgravity) [reply shared/build.gradle.kts:30]: So we will have to wait for the ios-template to fix it first? 🙏

Comment (phongvhd93) [reply shared/build.gradle.kts:30]: solved in [2b6549e](https://github.com/nimblehq/kmm-templates/pull/19/commits/2b6549e43b56f9a5f62ae0dec7b33a4c59691359)

**Review by luongvo — APPROVED** [review]

**Comment by luongvo** [comment]
@markgravity One more approval? 👋

**Review by markgravity — APPROVED** [review]

**Comment by luongvo** [comment]
@phongvhd93 it's ready for merging now. Let's go faster :rocket:

**Pull request merged**